ArcadiaParties.API/Controllers/UsersController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using ArcadiaParties.CQRS.Queries;
+using ArcadiaParties.Data.Abstractions.DTOs;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ArcadiaParties.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {        
+        private readonly IMediator _mediator;
+
+        public UsersController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [ProducesResponseType(typeof(IEnumerable<UserDTO>),StatusCodes.Status200OK)]
+        [SwaggerOperation(
+             Summary = "Returns all users"
+        )]
+        [Authorize(Roles = "Administrator")]
+        [HttpGet]
+        [Route("GetUsers")]
+        public async Task<IActionResult> GetUsers(CancellationToken cancellationToken)
+        {
+            var query = new GetAllUsersQuery();
+            var users = await _mediator.Send(query, cancellationToken);
+
+            return Ok(users);
+        }
+
+        [ProducesResponseType(typeof(UserDTO), StatusCodes.Status200OK)]
+        [SwaggerOperation(
+             Summary = "Returns current authenticated user"
+        )]
+        [Authorize]
```

Comment:
You decided to use `Roles` in each `Authorize` attribute.

Reply:
I've changed middleware. Now it returns 403 status code, if there is no current user in database